### PR TITLE
Add event

### PR DESCRIPTION
**Pull request by Justus-Saringer**

### CODE REVIEW

File: demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        SubtractButton {
+            number--
+        }
+    }
+
+}
+
+@Composable
+private fun EventText(number: String) {
```
Comment (Darius3001) [inline demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt:80]: What did you intend to show here?:

File: demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            color = White
+        )
+    }
+}
+
+@Composable
+private fun AddButton(onClick: () -> Unit) {
+    // if the child is clicked, an event is sent to the parents.
+    Button(
+        onClick = { onClick() },
```
Comment (Darius3001) [inline demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt:101]: You could also just do `onClick = onClick`

File: demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        modifier = Modifier.padding(10.dp)
+    ) {
+        Text(text = "ADD +1")
+    }
+}
+
+@Composable
+private fun SubtractButton(onClick: () -> Unit) {
+    Button(
+        onClick = { onClick() }
```
Comment (Darius3001) [inline demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt:111]: same here

File: demo/src/main/java/de/check24/compose/demo/features/events/EventFragment.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import de.check24.compose.demo.databinding.EventLayoutBinding
+
+class EventFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = EventFragment()
```
Comment (Darius3001) [inline demo/src/main/java/de/check24/compose/demo/features/events/EventFragment.kt:14]: Do you need this? If you want a new instance you cant just do `EventFragment()`

Comment (Darius3001) [reply demo/src/main/java/de/check24/compose/demo/features/events/EventFragment.kt:14]: i also dont see where you use this

Comment (Justus-Saringer) [reply demo/src/main/java/de/check24/compose/demo/features/events/EventFragment.kt:14]: You are right. I don't need it.

I use the Fragment in the Activity. I instantiate it there.

File: demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt (hunk 1 above)
Comment (Justus-Saringer) [reply demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt:80]: I don't know what you mean. Should I rename the composable?

File: code/events/Events.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+}
+
+@Composable
+fun AddButton(onClick: () -> Unit) {
+    // if the child is clicked, an event is sent to the parents.
+    Button(
+        onClick = { onClick() },
+        modifier = Modifier.padding(10.dp)
+    ) {
```
Comment (Darius3001) [inline code/events/Events.kt:56]: dont forget to adapt changes to website

File: demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt (hunk 1 above)
Comment (Darius3001) [reply demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt:80]: Nono i just dont understand what the composable is for. it says eventtext but i dont see any event

Comment (Justus-Saringer) [reply demo/src/main/java/de/check24/compose/demo/features/events/ComposableEventsActivity.kt:80]: Ah, alright. I will change the naming. 
It just shows the number, which will be increased or decreased by the button events.

**Review by Darius3001 — APPROVED** [review]

**Pull request merged**